src/python-lxc/lxc/__init__.py

```diff
@@ -217,22 +217,18 @@ def create(self, template=None, flags=0, args=()):
             "args" (optional) is a tuple of arguments to pass to the
             template. It can also be provided as a dict.
         """
-
         if isinstance(args, dict):
-            template_args = []
+            tmp_args = []
             for item in args.items():
-                template_args.append("--%s" % item[0])
-                template_args.append("%s" % item[1])
-        else:
-            template_args = args
-
-        if template:
-            return _lxc.Container.create(self, template=template,
-                                         flags=flags,
-                                         args=tuple(template_args))
-        else:
-            return _lxc.Container.create(self, flags=flags,
-                                         args=tuple(template_args))
+                tmp_args.append("--%s" % item[0])
+                tmp_args.append("%s" % item[1])
+        template_args = {}
+        template_args['template'] = template
```

Comment:
This is now always passing template to Create instead of only passing it if it's set.